lib/galaxy/tools/parameters/validation.py

```diff
@@ -248,6 +249,23 @@ def validate( self, value, trans=None ):
                 raise ValueError( self.message )
 
 
+class DatasetFilesPathEmptyValidator( Validator ):
+    """Validator that checks if a dataset's files_path exists and is not empty."""
+    def __init__( self, message=None ):
+        self.message = message
+
+    @classmethod
+    def from_element( cls, param, elem ):
+        return cls( elem.get( 'message', None ) )
+
+    def validate( self, value, trans=None ):
+        if value:
+            if not os.path.isdir(value.extra_files_path) or len(os.listdir(value.extra_files_path)) == 0:
```

Comment:
The code in question is this (set_total_size is being called by get_total_size):
```
    def set_total_size( self ):
        if self.file_size is None:
            self.set_size()
        self.total_size = self.file_size or 0
        if self.object_store.exists(self, extra_dir=self._extra_files_path or "dataset_%d_files" % self.id, dir_only=True):
            for root, dirs, files in os.walk( self.extra_files_path ):
                self.total_size += sum( [ os.path.getsize( os.path.join( root, file ) ) for file in files if os.path.exists( os.path.join( root, file ) ) ] )
```

So `if value.get_total_size() == value.get_size()` should do what you want, right?